static/app/actionCreators/tags.tsx

```diff
@@ -192,3 +208,214 @@ export function fetchSpanFieldValues({
     query,
   });
 }
+
+type FetchOrganizationTagsParams = {
+  orgSlug: string;
+  // TODO: Change this to Dataset type once IssuePlatform is added
+  dataset?: string;
+  projectIds?: string[];
+  useCache?: boolean;
+};
+
+export const makeFetchOrganizationTags = ({
+  orgSlug,
+  dataset,
+  projectIds,
+  useCache = true,
+}: FetchOrganizationTagsParams): ApiQueryKey => [
+  `/organizations/${orgSlug}/tags/`,
+  {query: {dataset, useCache, project: projectIds}},
+];
+
+export const useFetchOrganizationTags = (
+  params: FetchOrganizationTagsParams,
+  options: Partial<UseApiQueryOptions<Tag[]>>
+) => {
+  return useApiQuery<Tag[]>(makeFetchOrganizationTags(params), {
+    staleTime: Infinity,
+    keepPreviousData: true,
+    ...options,
+  });
+};
+
+type UseFetchIssueTagsParams = {
+  orgSlug: string;
+  projectIds: string[];
+  useCache?: boolean;
+};
+
+export const useFetchIssueOrganizationTags = ({
```

Comment:
Files in `actionCreators` should ideally be reusable across the app, so I'd suggest moving this hook to `views/issueList` since it should only be used by the components in that directory.